### PR TITLE
Add Hive and Swap.Signa

### PR DESCRIPTION
**Pull request by Valareos**
This will, like Wsigna and WBNB did, extend the Signum Network Crosschain capabilities.

Swap.signa exists already on Hive Engine, and there is a Hive/Swap.Signa LP Pool

### CODE REVIEW

**Comment by Valareos** [comment]
#214

**Comment by jjos2372** [comment]
Hello, thanks for the PR. I don't see any address validation on it, this way any address (not empty) will be accepted. We should have some logic checking for valid addresses, this avoids mixing up different chains and other user mistakes that will lead to excessive disputes.

For example, please check https://github.com/btdex/btdex/blob/f2fe936f08767188df49c448c15fc21ecdc8555d/src/main/java/btdex/markets/MarketDOGE.java#L45

File: src/main/java/btdex/markets/MarketSSIGNA.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	}
+	
+	@Override
+	public long getID() {
+		return MARKET_SSIGNA;
+	}
+	
+	@Override
+	public int getUCA_ID() {
+		return 10776;
```
Comment (jjos2372) [inline src/main/java/btdex/markets/MarketSSIGNA.java:31]: Is this correct?

Comment (rogerwim) [reply src/main/java/btdex/markets/MarketSSIGNA.java:31]: > Is this correct?
What are you referring to?

**Comment by rogerwim** [comment]
> Hello, thanks for the PR. I don't see any address validation on it, this way any address (not empty) will be accepted. We should have some logic checking for valid addresses, this avoids mixing up different chains and other user mistakes that will lead to excessive disputes.
> 
> For example, please check
> 
> https://github.com/btdex/btdex/blob/f2fe936f08767188df49c448c15fc21ecdc8555d/src/main/java/btdex/markets/MarketDOGE.java#L45

Any address is valid in hive, as long as it starts with a @, so this is pretty much impossible

**Comment by jjos2372** [comment]
> Any address is valid in hive, as long as it starts with a @, so this is pretty much impossible

Could you please provide a reference/link for this requirement. And if that is the case, it should check for the starting @ and any other limitation (like charset, white spaces, etc).

File: src/main/java/btdex/markets/MarketSSIGNA.java (hunk 1 above)
Comment (jjos2372) [reply src/main/java/btdex/markets/MarketSSIGNA.java:31]: That is the SIGNA UCA_ID, not the wrapped one. But I guess we will not get one different anyway (it was similar to WSIGNA).

**Pull request merged**